### ISSUE TITLE
Errors due to /pw reload

### ISSUE THREAD

**Issue by SlimeDog** [PyroTempus/My-Plugins#42]
PWP 3.3.0

```
> pw reload
[17:17:15] [Server thread/INFO]: [PyroWeatherPro] Disabling PyroWeatherPro v3.3.0
[17:17:15] [Server thread/ERROR]: Error occurred while disabling PyroWeatherPro v3.3.0 (Is it up to date?)
java.lang.NullPointerException: null
        at me.arsmagica.WeatherTypes.GammaRayBursts.stopGammaRayBurst(GammaRayBursts.java:173) ~[?:?]
        at me.arsmagica.PyroWeather.onDisable(PyroWeather.java:140) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:267) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.plugin.java.JavaPluginLoader.disablePlugin(JavaPluginLoader.java:371) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.plugin.SimplePluginManager.disablePlugin(SimplePluginManager.java:429) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at me.arsmagica.Extra.CommandsMain.onCommand(CommandsMain.java:165) ~[?:?]
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:46) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:141) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at org.bukkit.craftbukkit.v1_13_R2.CraftServer.dispatchCommand(CraftServer.java:704) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.PlayerConnection.handleCommand(PlayerConnection.java:1621) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.PlayerConnection.a(PlayerConnection.java:1461) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:45) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:1) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:9) ~[spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at net.minecraft.server.v1_13_R2.SystemUtils.a(SourceFile:199) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:896) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:417) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:831) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:729) [spigot-1.13.2.jar-2019-04-22-2230:git-Spigot-1a3504a-a46fdbc]
        at java.lang.Thread.run(Thread.java:834) [?:?]
```
```
[17:17:15] [Server thread/INFO]: [PyroWeatherPro] Enabling PyroWeatherPro v3.3.0
[17:17:15] [Server thread/WARN]: Enabled plugin with unregistered PluginClassLoader PyroWeatherPro v3.3.0
[17:17:15] [Server thread/WARN]: [LagMonitor] Plugin PyroWeatherPro is performing a blocking I/O operation (Socket: https://api.spigotmc.org/legacy/premium.php?user_id=56363&resource_id=60394&nonce=-1895503016) on the main thread. This could affect the server performance, because the thread pauses until it gets the response. Such operations should be performed asynchronous from the main thread. If this happens on server startup, fixing it can drastically reduce startup time. Keep in mind to keep the code thread-safe. Report it to the plugin author
[17:17:15] [Server thread/WARN]: [LagMonitor] Source: PyroWeather.java, method loadConfig0, line -1
```
```
[17:17:15] [Server thread/WARN]: [LagMonitor] Plugin PyroWeatherPro is performing a blocking I/O operation (Socket: https://api.spigotmc.org/legacy/update.php?resource=60394) on the main thread. This could affect the server performance, because the thread pauses until it gets the response. Such operations should be performed asynchronous from the main thread. If this happens on server startup, fixing it can drastically reduce startup time. Keep in mind to keep the code thread-safe. Report it to the plugin author
[17:17:15] [Server thread/WARN]: [LagMonitor] Source: DisableWeather.java, method a, line 63
[17:17:15] [Server thread/WARN]: java.io.IOException: Connect failed
[17:17:15] [Server thread/WARN]:        at me.lucko.networkinterceptor.interceptors.ProxySelectorInterceptor$LoggingSelector.select(ProxySelectorInterceptor.java:59)
[17:17:15] [Server thread/WARN]:        at com.github.games647.lagmonitor.listener.BlockingConnectionSelector.select(BlockingConnectionSelector.java:33)
[17:17:15] [Server thread/WARN]:        at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1175)
[17:17:15] [Server thread/WARN]:        at java.base/sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1065)
[17:17:15] [Server thread/WARN]:        at java.base/sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1063)
[17:17:15] [Server thread/WARN]:        at java.base/java.security.AccessController.doPrivileged(Native Method)
[17:17:15] [Server thread/WARN]:        at java.base/java.security.AccessController.doPrivilegedWithCombiner(AccessController.java:793)
[17:17:15] [Server thread/WARN]:        at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1062)
[17:17:15] [Server thread/WARN]:        at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177)
[17:17:15] [Server thread/WARN]:        at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1581)
[17:17:15] [Server thread/WARN]:        at java.base/sun.net.www.protocol.http.HttpURLConnection$9.run(HttpURLConnection.java:1501)
[17:17:15] [Server thread/WARN]:        at java.base/sun.net.www.protocol.http.HttpURLConnection$9.run(HttpURLConnection.java:1499)
[17:17:15] [Server thread/WARN]:        at java.base/java.security.AccessController.doPrivileged(Native Method)
[17:17:15] [Server thread/WARN]:        at java.base/java.security.AccessController.doPrivilegedWithCombiner(AccessController.java:793)
[17:17:15] [Server thread/WARN]:        at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1498)
[17:17:15] [Server thread/WARN]:        at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:245)
[17:17:15] [Server thread/WARN]:        at me.arsmagica.Listeners.DisableWeather.a(DisableWeather.java:63)
[17:17:15] [Server thread/WARN]:        at me.arsmagica.PyroWeather.onEnable(PyroWeather.java:119)
[17:17:15] [Server thread/WARN]:        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:265)
[17:17:15] [Server thread/WARN]:        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:347)
[17:17:15] [Server thread/WARN]:        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:410)
[17:17:15] [Server thread/WARN]:        at me.arsmagica.Extra.CommandsMain.onCommand(CommandsMain.java:166)
[17:17:15] [Server thread/WARN]:        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:46)
[17:17:15] [Server thread/WARN]:        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:141)
[17:17:15] [Server thread/WARN]:        at org.bukkit.craftbukkit.v1_13_R2.CraftServer.dispatchCommand(CraftServer.java:704)
[17:17:15] [Server thread/WARN]:        at net.minecraft.server.v1_13_R2.PlayerConnection.handleCommand(PlayerConnection.java:1621)
[17:17:15] [Server thread/WARN]:        at net.minecraft.server.v1_13_R2.PlayerConnection.a(PlayerConnection.java:1461)
[17:17:15] [Server thread/WARN]:        at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:45)
[17:17:15] [Server thread/WARN]:        at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:1)
[17:17:15] [Server thread/WARN]:        at net.minecraft.server.v1_13_R2.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:9)
[17:17:15] [Server thread/WARN]:        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[17:17:15] [Server thread/WARN]:        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[17:17:15] [Server thread/WARN]:        at net.minecraft.server.v1_13_R2.SystemUtils.a(SourceFile:199)
[17:17:15] [Server thread/WARN]:        at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:896)
[17:17:15] [Server thread/WARN]:        at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:417)
[17:17:15] [Server thread/WARN]:        at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:831)
[17:17:15] [Server thread/WARN]:        at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:729)
[17:17:15] [Server thread/WARN]:        at java.base/java.lang.Thread.run(Thread.java:834)
```
The last one appears to be trying to make a network call. Check-updates is disabled in config.yml

**Comment by PyroTempus** [comment]
Yeah. The last one makes a call to the Spigot API to grab the current resource version. (It's a slight attempt to stop piracy. I'll fix it making errors when used in an offline environment. 
`[17:17:15] [Server thread/WARN]: [LagMonitor] Source: PyroWeather.java, method loadConfig0, line -1`
I'm pretty sure this is something that I can't fix although I'm unsure. I've never seen it happen. loadConfig0 is something on Spigots end to do with protection I believe. I'm not entirely sure on this one though. It does say line -1 and there isn't a method called that anywhere in my plugin. From what I know, it automatically injects it into the resource.

**Comment by SlimeDog** [comment]
> Yeah. The last one makes a call to the Spigot API to grab the current resource version. (It's a slight attempt to stop piracy. I'll fix it making errors when used in an offline environment.

```
GeneralSettings:
...
  VersionChecker: false
```

**Comment by PyroTempus** [comment]
It's not a brilliant way to stop it but it's a decent attempt by stopping out of date versions from enabling. I forgot to add an exception grabber which makes it so errors are spat into console if it can't make that connection which is something I need to fix on my end.

**Comment by SlimeDog** [comment]
Thanks.

**Comment by PyroTempus** [comment]
Errors fixed for v3.3.1.
